WooCommerce/Classes/Yosemite/TrackEventRequestNotificationHandler.swift

```diff
@@ -0,0 +1,58 @@
+import Foundation
+
+final class TrackEventRequestNotificationHandler {
+
+    /// NotificationCenter Tokens for tracking Analytics event
+    ///
+    private var trackingObservers: [NSObjectProtocol]?
+
+    /// NotificationCenter
+    ///
+    private let notificationCenter: NotificationCenter
+
+    /// Analytics
+    ///
+    private let analytics: Analytics
+
+    init(notificationCenter: NotificationCenter = .default,
+         analytics: Analytics = ServiceLocator.analytics) {
+        self.notificationCenter = notificationCenter
+        self.analytics = analytics
+        startListeningToNotifications()
+    }
+}
+
+private extension TrackEventRequestNotificationHandler {
+    /// Starts listening for Notifications
+    ///
+    func startListeningToNotifications() {
+        let newPasswordCreatedObserver = notificationCenter.addObserver(forName: .ApplicationPasswordsNewPasswordCreated,
+                                                                        object: nil,
+                                                                        queue: .main) { [weak self] note in
```

Comment:
```swift
open func addObserver(forName name: NSNotification.Name?, object obj: Any?, queue: OperationQueue?, using block: @escaping @Sendable (Notification) -> Void) -> NSObjectProtocol
```
I am afraid that we have to manually pass the queue.  

BTW, I got this from [here](https://github.com/woocommerce/woocommerce-ios/blob/trunk/WooCommerce/Classes/Yosemite/AuthenticatedState.swift#L147). Kindly let me know if this can be improved further.